File: cpp/ISP/step2/Car.cpp
#pragma once
#include "Vehicle.cpp"

class Car : public Vehicle {
public:
	bool IsRadioOn() {
		return radio_on_;
	}

	void TurnRadioOn() override {
		radio_on_ = true;
	}

	void TurnRadioOff() override {
		radio_on_ = false;
	}

	void TurnCameraOn() override {
		// nothing to do here
	}

	void TurnCameraOff() override {
		// nothing to do here
	}

private:
	bool radio_on_;
};

File: python/ISP/step2/car.py
from vehicle import Vehicle

class Car(Vehicle):
    def __init__(self) -> None:
        super().__init__()
        self.__radio_on = False

    def is_radio_on(self) -> bool:
        return self.__radio_on

    def turn_radio_on(self):
        self.__radio_on = True

    def turn_radio_off(self):
        self.__radio_on = False

    def turn_camera_on(self):
        # nothing to do here
        pass

    def turn_camera_off(self):
        # nothing to do here
        pass
